pcdscalc/misc_calcs.py

```diff
@@ -0,0 +1,351 @@
+import numpy as np
+from .be_lens_calcs import alias
+
+# TODO: find a better number
+_manual_energy = 9.8
+
+
+def set_energy(energy):
+    """
+    Set a global, manual, working energy for doing offline calcs.
+    """
+    global _manual_energy
+    _manual_energy = energy
+
+
+def check_id(material_id):
+    """Check to see if you are using an alias. Return the chemical formula."""
+    try:
+        return alias[material_id]
+    except Exception:
+        return material_id
+
+
+def cosd(angle):
+    """Cos of an angle specified in degrees."""
+    arad = np.deg2rad(angle)
+    x = np.cos(arad)
+    return x
+
+
+def sind(angle):
+    """Sin of an angle specified in degrees."""
+    arad = np.deg2rad(angle)
+    x = np.sin(arad)
+    return x
+
+
+def asind(angle):
+    """Arcsin in degrees. Closed interval of [-pi/2, pi/2]."""
+    a = np.arcsin(angle)
+    a = np.rad2deg(a)
+    return a
+
+
+def get_energy(energy=None, correct_ev=True):
+    """
+    Get working energy.
+
+    Convert to eV if correct_ev == True
+    If energy passed in, return it, otherwise return manual_energy if set, or
+    machine energy otherwise.
+
+    Parameters
+    ----------
+    energy : number
+        Photon energy in eV or keV.
+    correct_ev : bool
+        Indicates if energy should be converted to eV.
+
+    Returns
+    -------
+    en : number
+        Photon energy in eV or keV.
+    """
+    en = None
+    if energy is not None:
+        en = energy
+    elif _manual_energy is not None:
+        en = _manual_energy
+    else:
+        # en = _ePv.get()
+        # TODO: we need a way of getting the actual energy here.
```

Comment:
I'm just not doing anything here because I'm not sure again how we want to get the actual energy - this should be fixed when we have a way of getting the current desired energy

Reply:
Didn't we come up with some way of setting default energy in the previous PR? Can't seem to find it.

Reply:
We talked about it in two different places, as well as on the chat maybe? I ended up not implementing anything because I think we decided that it should be a different Issue if I remember correctly. 

https://github.com/pcdshub/pcdscalc/pull/8#discussion_r509642283

https://github.com/pcdshub/pcdsdevices/pull/633#discussion_r510578790

The main idea was this:
```
photon_ev_hxr.subscribe(_set_current_energy)

def get_energy():
   if user_specified_energy is not None:
       return user_specified_energy
    return _current_energy

def other_calculation_function(, ..., energy=None):
    if energy is None:
        energy = get_energy()
```
which is similar to what I have now (not really but similar idea) except I don't have the hook for getting the _curren_energy.
I need to go back and think about it again...

Reply:
In the old code they were taking this directly from a PV (in calcs), but we did not want to do this here.
Instead I think we wanted to have this in `pcdsdevices` or somewhere else and let the devices that need these calculations get the energy they want.

So I think I should just not do anything in these calculations with Energy - i probably should not even have `get_energy()` and `set_energy()` here at all, instead I should expect the users to send the proper energy when using these calculations.

Reply:
I should however open a PR maybe for this idea (https://github.com/pcdshub/pcdscalc/pull/8#discussion_r509793381)??

Reply:
Yeah, seems like we should solve this generally once and for all. Seems fine to put this in a separate PR after getting these calculation functions in.

Reply:
When we set energy, we only set virtual motors and keep the user values and offsets intact, this allows us track the drift of the system over time (as the difference between the virtual/pseudo motor values drift away from each other). Only the Psuedo motors's values are use for energy change and reset.

In current (old python) version, get_energy calcuate energy directly from th1C.  th1 and th1C was set to identical nominal values after the first run 18 alignment/commissioning.